### PR TITLE
Auth Review Changes

### PR DESCRIPTION
**Pull request by mpscholten**
@ianobermiller can you take a look?

### CODE REVIEW

File: thin-backend-react/auth-ui.tsx [hunk 1]
```diff
@@ -31,30 +33,31 @@ export function LoginAndSignUp() {
 
 interface LoginProps {
     description?: string,
-    onSignUpClick: () => void
+    onSignUpClick: () => void,
+    appIcon?: string
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:37]: Add a doc-block comment saying what size this icon should be. Actually, I suggest replacing this with a `ReactNode` so you can provide your own `img` element with a size and such. Also if you want to use an optimizing image component like `@next/image`.

File: thin-backend-react/auth-ui.tsx [hunk 2]
```diff
@@ -31,30 +33,31 @@ export function LoginAndSignUp() {
...
-                <div className="thin-login-box">
-                    <div className="thin-login-icon-container">
-                        <img src="https://testvercelreview.thinbackend.app/thin-icon-black.png"/>
+export function Login({ description, onSignUpClick, appIcon = DEFAULT_APP_ICON }: LoginProps) {
+    return <div className="thin-auth">
+        <div className="thin-auth-container">
+            <div className="thin-auth-container-inner">
+                <div className="thin-auth-box">
+                    <div className="thin-auth-icon-container">
+                        <img src={appIcon}/>
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:45]: This should include the size to avoid layout thrash.

File: thin-backend-react/auth-ui.tsx [hunk 3]
```diff
@@ -228,7 +232,7 @@ function SignUpForm({ description, onLoginClick }) {
     return <form method="POST" action="#" onSubmit={onSubmit}>
         <h1>Sign Up</h1>
 
-        <p className="thin-login-description">
+        <p className="thin-auth-description">
             {description || 'Sign up for an account to use this application.'}
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:236]: use a default when destructuring instead

File: thin-backend-react/auth-ui.tsx [hunk 4]
```diff
@@ -124,7 +127,7 @@ function LoginForm() {
         <div>
             <button type="submit" disabled={isLoading}>
                 {isLoading
-                    ? <span className="thin-login-spinner-border thin-login-spinner-border-sm" role="status" aria-hidden="true"></span>
+                    ? <LoadingSpinner small/>
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:130]: consider allowing the spinner to be replaced via props, as apps will often have their own style of spinner.

File: thin-backend-react/auth-ui.tsx [hunk 5]
```diff
@@ -267,14 +271,14 @@ function SignUpForm({ description, onLoginClick }) {
...
                     : "Sign Up"
                 }
             </button>
         </div>
 
         <p>
-            <span className="thin-login-signup">Already have an account?</span> <a href={ihpBackendUrl('/NewUser')} onClick={() => { event.preventDefault(); onLoginClick(); }}>Login</a>
+            <span className="thin-auth-signup">Already have an account?</span> <a href={ihpBackendUrl('/NewUser')} onClick={() => { event.preventDefault(); onLoginClick(); }}>Login</a>
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:281]: Will this work without JS? That's pretty neat.

File: thin-backend-react/auth-ui.tsx [hunk 6]
```diff
@@ -351,4 +355,12 @@ function useEmailConfirmationParameters() {
     }, []);
 
     return [emailConfirmationParameters, setEmailConfirmationParameters];
+}
+
+interface LoadingSpinnerProps {
+    small?: boolean;
+}
+function LoadingSpinner({ small = false }: LoadingSpinnerProps) {
+    const className = 'thin-auth-spinner-border' + (small ? ' thin-auth-spinner-border-sm' : '');
```
Comment (ianobermiller) [inline thin-backend-react/auth-ui.tsx:364]: consider using https://www.npmjs.com/package/classcat or clsx

**Comment by ianobermiller** [comment]
Overall these changes look good, sorry for my delay in taking a look. One other thing I'd suggest is to leave the `input` elements uncontrolled (don't use `useState` at all) and instead read them out of the form data in the onSubmit event. It is slightly less type-safe, but vastly more efficient and less code. This component is small enough that it really doesn't matter, so it's just a suggestion.

File: thin-backend-react/auth-ui.tsx (hunk 5 above)
Comment (mpscholten) [reply thin-backend-react/auth-ui.tsx:281]: Basically yes, but if you have no JS enabled, the app might not correctly handle the redirect back from the thin login page (Thin redirects back to the SPA url and passes `userId` and a random string `accessToken` via query parameters. These then need to be exchanged for a JWT via an HTTP call)

**Pull request merged**